src/mixins/canvas_events.mixin.js

```diff
@@ -271,10 +271,8 @@
       if (target) {
         target.isMoving = false;
       }
-
-      shouldRender && this.renderAll();
-
       this._handleCursorAndEvent(e, target);
```

Comment:
why mouseup is managed different from other events? it gets its own function to fire the events.